docarray/typing/abstract_type.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional, Type, TypeVar, Union
+
+from pydantic import BaseConfig
+from pydantic.fields import ModelField
+
+from docarray.document import BaseNode
+
+T = TypeVar("T")
+
+
+class AbstractType(BaseNode):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
```

Comment:
If it is abstract it shouldn't offer this implementation, right? And inherit from `ABC`?

Reply:
I think it is okay to offer this implementation since this is a common pattern even though it is abstract. An abstract class mean only one of method is abstract not all of them. ABC makes sense though